caddyhttp/proxy/upstream.go

```diff
@@ -42,6 +43,7 @@ type staticUpstream struct {
 		Interval time.Duration
 		Timeout  time.Duration
 		Host     string
+		Port     int
```

Comment:
Why do you use an `int`? If you use a `string` you do not need to convert it back and forth